lib/autoupdate/agent/updater.go

```diff
@@ -341,18 +353,53 @@ func (u *Updater) Install(ctx context.Context, override OverrideConfig) error {
 	// Only write the configuration file if the initial update succeeds.
 	// Note: skip_version is never set on failed enable, only failed update.
 
-	if err := writeConfig(u.ConfigPath, cfg); err != nil {
+	if err := writeConfig(u.UpdateConfigPath, cfg); err != nil {
 		return trace.Wrap(err, "failed to write %s", updateConfigName)
 	}
 	u.Log.InfoContext(ctx, "Configuration updated.")
 	return trace.Wrap(u.notices(ctx))
 }
 
+// findAgentProxy finds a proxy in teleport.yaml if not specified or set in update configuration.
+// Note that any implicitly defaulted port in teleport.yaml is explicitly defaulted (to 3080) by this method.
+func (u *Updater) findAgentProxy(ctx context.Context) string {
+	fc, err := config.ReadFromFile(u.TeleportConfigPath)
+	if err != nil {
+		u.Log.DebugContext(ctx, "Unable to find proxy in Teleport config", "config", u.TeleportConfigPath, errorKey, err)
+		return ""
+	}
+	addr, err := libutils.ParseHostPortAddr(fc.ProxyServer, libdefaults.HTTPListenPort)
```

Comment:
Note: in config v1 and v2 if the proxy_server is not set, teleport uses the `auth_servers` first element as a proxy server (this is not true in config v3, we split auth_servers into `proxy_server` and `auth_server`). It might be a good thing to try looking up the proxy url form this field as well as many users are still relying on our old config format.